src/pkg/registry/client.go

```diff
@@ -379,6 +411,45 @@ func (c *client) PushBlob(repository, digest string, size int64, blob io.Reader)
 	return c.monolithicBlobUpload(location, digest, size, blob)
 }
 
+// PushBlobChunk pushes the specified blob, but by chunked, refer to https://github.com/opencontainers/distribution-spec/blob/main/spec.md#push for more details.
+func (c *client) PushBlobChunk(repository, digest string, blobSize int64, chunk io.Reader, start, end int64, location string) (string, error) {
+	var err error
+	// first chunk need to initialize blob upload location
+	if start == 0 {
+		location, _, err = c.initiateBlobUpload(repository)
+		if err != nil {
+			return location, err
+		}
+	}
+
+	lastChunk := end == blobSize
```

Comment:
Does `start` start from 0? I mean that if a blob is only 2 bytes, what are the values of `start`, `end` and `blobSize`? Is `0`, `1`, `2` or `0`, `2`, `2`?

Reply:
Yes, the start should from 0, if the blob size is 2 bytes, and if the chunk size greater than 2 bytes, then the request will fallback to copy by blob, if the chunk size less than blob size, such as 1 byte, the 1st chunk is start(0), end(1), blobSize(2), the 2st chunk is start(2), end(2), blobSize(2).

Reply:
Please double check this part as we discussed offline